### ISSUE TITLE
Add DB2 to aws_dms_endpoint engine_name

### ISSUE THREAD

**Issue by rtaylor30** [terraform-providers/terraform-provider-aws#7312]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When attempting to add in a new DMS Endpoint, for DB2, it gives a validation error.

```
resource "aws_dms_endpoint" "db2-source" {
  endpoint_id = "db2-source"
  endpoint_type = "source"
  engine_name = "db2"
  database_name = "sample"
  username = "sampleuser"
  password = "samplepassword"

  tags {
    Name = "db2-sample-endpoint"
  }
}
```

After running `terraform apply`

```
Error: aws_dms_endpoint.db2-source: expected engine_name to be one of [mysql oracle postgres dynamodb mariadb aurora aurora-postgresql redshift sybase sqlserver mongodb s3 azuredb], got db2
```

AWS has added the ability to use DB2 as a source, found in the documentation for AWS DMS Endpoint.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dms-endpoint.html#cfn-dms-endpoint-enginename

### New or Affected Resource(s)

* Terraform v0.11.11
  * provider.aws v1.56.0

### References

* https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dms-endpoint.html#cfn-dms-endpoint-enginename

**Comment by kireledan** [comment]
Created a PR for this issue #9097

**Comment by bflad** [comment]
Support for the `db2` engine name has been merged and will release with version 2.17.0 of the Terraform AWS Provider, likely later this week. 👍 Thanks, @kireledan!

**Comment by bflad** [comment]
This has been released in [version 2.17.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2170-june-28-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
